swift/ql/lib/codeql/swift/elements/type/TupleType.qll

```diff
@@ -6,4 +6,13 @@ private import codeql.swift.generated.type.TupleType
  * (Int, String)
  * ```
  */
-class TupleType extends Generated::TupleType { }
+class TupleType extends Generated::TupleType {
+  /**
+   * Gets the type that is this tuple type with any element labels removed.
+   */
+  TupleType asUnlabeled() {
+    result.getNumberOfTypes() = this.getNumberOfTypes() and
+    not exists(int index | result.getType(index) != this.getType(index)) and
+    not exists(result.getName(_))
+  }
```

Comment:
Two questions:
- This looks like something that could blow up with a bad join order, right? That is, it we start by joining together all `TupleTypes` that has the same number of types. Does the joins look okay in this case?
- Does the unlabeled version of a tuple type always exist? Or should the `QLDoc` say `[...], if any.`? And if this predicate really is partial: Is that a problem for the way we use it in `stripType`?